plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -400,14 +402,15 @@ public class DeltaLakeMetadata
     private final boolean deleteSchemaLocationsFallback;
     private final boolean useUniqueTableLocation;
     private final boolean allowManagedTableRename;
-    private final Map<SchemaTableName, Long> queriedVersions = new ConcurrentHashMap<>();
+    private final Map<QueriedTable, Long> queriedVersions = new ConcurrentHashMap<>();
     private final Map<QueriedTable, TableSnapshot> queriedSnapshots = new ConcurrentHashMap<>();
```

Comment:
Do `QueriedTable` have different meanings in these maps? In `queriedVersions` `QueriedTable` identifies a table optionally at a version. In `queriedSnapshots` it is a table at a version. Do we have a guarantee that `QueriedVersion.version().isPresent()` for all entries of `queriedSnapshots`?

Reply:
@jkylling yes, the current structure is containing unneeded redundancy.
If we're dealing with the query `SELECT * FROM default.t FOR VERSION AS OF 9` then we'll have in `queriedVersions` a mapping between `<QueriedTable('default.t', Optional.of(9)), 9>`.

We need actually just to know what is the "latest" table version as in the situation where the version specified is `Optional.empty()` - corresponds to `SELECT * FROM t`

One option to take into consideration would be to have:
```
    private final Map<SchemaTableName, Long> tablesLatestVersions = new ConcurrentHashMap<>();
    private final Map<QueriedTable, TableSnapshot> queriedSnapshots = new ConcurrentHashMap<>();

    private record QueriedTable(SchemaTableName schemaTableName, long version)
```